code/modules/projectiles/gun.dm

```diff
@@ -344,9 +344,7 @@
 
 	update_held_icon()
 
-	//update timing
-	var/delay = max(burst_delay+1, fire_delay)
-	user.setClickCooldown(min(delay, DEFAULT_QUICK_COOLDOWN))
+	user.setClickCooldown(fire_delay)
```

Comment:
```suggestion
	user.setClickCooldown(max(burst_delay+1, fire_delay))
```
As discussed on discord. You are removing the burst delay from the equation, which will enable a user to squeeze of shots faster than it should be possible in a otherwise slower burst firemode